lib/hackmode/realmode.js

```diff
@@ -903,32 +951,51 @@ class N3hRealMode extends N3hMode {
       throw new Error('cannot _initDht without agentId string')
     }
     // Setup init data
-    const dhtInitOptions = { thisPeer: DhtEvent.peerHoldRequest(
-      agentId,
-      'hc://' + this._p2p.getId(),
-      Buffer.from('self').toString('base64'),
-      Date.now())
+    const dhtInitOptions = {
+      thisPeer: DhtEvent.peerHoldRequest(
+        agentId,
+        this.toDnaPeerTransport(this._p2p.getId()),
+        Buffer.from('self').toString('utf8'),
```

Comment:
hehe : ) this line converts a ucs2 string into a utf8 byte buffer, then parses that buffer from utf8 back into a ucs2 string.

can just write `'self'` here.

Reply:
Silly me.
The event type actually requires a base64 string for this field.